### DISCUSSION TITLE
Build fails when bundling PNGs

### DISCUSSION THREAD

**Discussion by paltze** [parcel-bundler/parcel#7740] [Q&A]
I am trying to make a game with Pixi.js with Parcel (v2.3.2). The `parcel src/index.html` command works fine but `parcel build src/index.html` fails with following error:
```bash
/c/Program Files/nodejs/npm: line 44:  1511 Illegal instruction     "$NODE_EXE" "$NPM_CLI_JS" "$@"
```

This is the exact command I am trying to use:
```bash
parcel build src/index.html --dist-dir build --no-source-maps
```

This is a sample JS that produces the same problem:
```js
import {Application, Texture, Sprite} from 'pixi.js';
import grass from './path/to/grass.png?quality=75';

const app = new Application();
document.body.appendChild(app.view);

const grassTexture = Texture.from(grass);
const grassSprite = Sprite.from(grassTexture);

app.stage.addChild(grassSprite);
```

This is the HTML:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Document</title>
    <script type="module" src="path/to/script.js"></script>
</head>
<body>
    
</body>
</html>
```
If I try to import PNG as WebP, the build works properly. I would have used WebP but it loses some colors.

On searching online I found a few solutions such as deleting `node_modules` and `package-lock.json` and reinstalling or running `npm rebuild`. I even uninstalled and reinstalled Node.js but nothing worked. Thanks in advance.

**Comment by Shinyaigeek** [comment]
hmm… in my environment, this works fine. Is there any context other than `index.html` and `script.js` ? (such as `.parcelrc`, or loaded png image…)

There is no basis, but I feel removing `.parcel-cache` may  resolve this issue.

**Reply by paltze** [reply]
No, removing `.parcel-cache` does not help. It was just a sample, but produced the same problem. This is my actual directory structure:
```
+---.parcel-cache
+---dist
+---src
|   +---assets
|   | ----floor.png
|   | ----floor.json
|   +---scenes
|   |   +---game
|   |   | ----map.ts
|   |   | ----sample.json
|   |   | ----socket.ts
|   |   ----game.ts
|   ----default.d.ts
|   ----index.html
|   ----main.ts
----package.json
----package-lock.json
----tsconfig.json
```
These are the contents of `tsconfig.json`:
```json
{
    "include":[
        "src/**/*"
    ]
}
```
These are the contents of package.json:
```json
{
  "private": true,
  "scripts": {
    "start": "parcel src/index.html --port 3001 --open",
    "build": "parcel build src/index.html --dist-dir build --no-source-maps"
  },
  "author": "paltze",
  "license": "ISC",
  "dependencies": {
    "pixi.js": "^6.2.2"
  },
  "devDependencies": {
    "parcel": "^2.3.2",
    "sharp": "^0.29.3"
  }
}
```

**Reply by Shinyaigeek** [reply]
can you share the github repository to reproduce?

**Reply by paltze** [reply]
It was a pain to use fonts, animations, physics with Pixi.js so I moved to Phaser. The problem still persists. This is the link to repo: https://github.com/paltze/demo.

**Reply by Shinyaigeek** [reply]
Thank you! but I cannot reproduce this... I feel this issue is caused by native module issue. https://github.com/parcel-bundler/parcel/issues/6113 https://github.com/parcel-bundler/parcel/issues/5773 these issues seems to be related.

**Reply by paltze** [reply]
Thank you for your time! Looks like like I will have to copy the PNG from results produced in `dist` folder.  ¯\_(ツ)_/¯
